server/src/index.ts

```diff
@@ -24,6 +24,17 @@ io.on("connection", (socket) => {
  console.log(`A user connected: ${socket.id}`);
 
  socket.on("join_room", (data) => {
+  
+  //Checking if there are any rooms currently connected to. If so looping through each room and users. If user connected to a room, leave the room. Also remove the user from the room sockets object
+  if(roomSockets){
+   for(const[prevRoom, users] of Object.entries(roomSockets)){
```

Comment:
Nice destructuring. So "prevRoom" is the room, and "users" are the users in those rooms? 
Then check if room has that socket.id like you commented above?